pkg/tpath/tpath.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpath
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+
+	"github.com/kylelemons/godebug/pretty"
+
+	"istio.io/operator/pkg/util"
+	"istio.io/pkg/log"
+)
+
+var (
+	// DebugPackage controls verbose debugging in this package. Used for offline debugging.
+	DebugPackage = false
```

Comment:
Can we use istio.io/istio/pkg/log scopes here? e.g.

```
var scope = log.Scope("tpath", "target path debugging")

scope.Debugf("log debug message")
```

Reply:
This is a good idea for the whole repo. I've created a issue: https://github.com/istio/istio/issues/15605